src/http.cpp

```diff
@@ -660,6 +656,47 @@ bool HttpReqUL::prepare(FileAccess* fa, const char* tempurl, SymmCipher* key,
     // unpad for POSTing
     out->resize(size);
 
+
+    const char *data = out->data();
+    byte c[12];
+    memset(c, 0, 12);
+
+#ifndef ALLOW_UNALIGNED_MEMORY_ACCESS
+    for (int l = size; l--;)
+    {
+        c[l % 12] ^= data[l];
+    }
+#else
```

Comment:
It could be interesting to perform a benchmarking to check if the code below is actually much faster than the version above.